zrml/prediction-markets/src/lib.rs

```diff
@@ -1646,6 +2012,17 @@ mod pallet {
         InvalidBaseAsset,
         /// A foreign asset in not registered in AssetRegistry.
         UnregisteredForeignAsset,
+        /// The early market close operation was not requested by the market creator.
+        RequesterNotCreator,
+        /// The early close would be scheduled after the original market period end.
+        EarlyCloseRequestTooLate,
+        /// This early close state is not valid.
+        InvalidEarlyCloseState,
+        /// There is no early close scheduled.
+        NoEarlyCloseScheduled,
+        /// After there was an early close already scheduled,
+        /// only the `CloseMarketsEarlyOrigin` can schedule another one.
+        OnlyAuthorizedCanScheduleEarlyClose,
```

Comment:
Errors should always be put at the ending of the error enum, otherwise we mess up error indices and give frontend devs a hard time.

Reply:
e8a85e4e742583d69e31521bec22c2e6d21887b6